core/spec/models/spree/order_shipping_method_spec.rb

```diff
@@ -0,0 +1,37 @@
+require 'spec_helper'
+
+describe Spree::Order do
+  let(:order) { create(:order_with_line_items) }
+  let(:quadcopter) { create(:shipping_method, name: "Quads'r'us", code: "QUADSRUS", cost: 200) }
+
+  it "preserves selected shipping method when update_cart and next! called" do
+    #choose more expensive quadcopter shipping method
+    original_shipment = order.shipments.first
+    quadcopter_rate = original_shipment.add_shipping_method(quadcopter)
+    original_shipment.selected_shipping_rate_id = quadcopter_rate.id
+    expect(order.shipments.first.shipping_method).to eq quadcopter
+
+    order.contents.update_cart({}) #contents don't really matter
+    order.next!
+    order.next! #takes to delivery state to create_proposed_shipments
+
+    expect(order.shipments.first.shipping_method).to eq quadcopter
+  end
+
+  let(:cheap_ups_cost) { 10 }
+
+  it "simplified case of api project's orders_controller_spec" do
```

Comment:
I found an additional test case. The jury is still out on whether it represents implicit assumptions by factories though. It's not passing now, but it's more focused than the api spec failure